src/uvutil.cpp

```diff
@@ -4,17 +4,44 @@
 
 
 class WriteOp {
-public:
-  uv_write_t handle;
+private:
   ExtendedWrite* pParent;
   uv_buf_t buffer;
+  // Bytes to write before writing the buffer
+  std::vector<char> prefix;
+  // Bytes to write after writing the buffer
+  std::vector<char> suffix;
+public:
+  uv_write_t handle;
 
   WriteOp(ExtendedWrite* parent, uv_buf_t data)
         : pParent(parent), buffer(data) {
     memset(&handle, 0, sizeof(uv_write_t));
     handle.data = this;
   }
 
+  void setPrefix(const char* data, size_t len) {
```

Comment:
Just to be extra safe:

```suggestion
  void setPrefix(const char* data, size_t len) {
    ASSERT_BACKGROUND_THREAD()
```